src/modules/app_gw/main.tf

```diff
@@ -182,6 +182,7 @@ resource "azurerm_monitor_diagnostic_setting" "app_gw" {
   name                       = "LogSecurity"
   target_resource_id         = azurerm_application_gateway.this.id
   log_analytics_workspace_id = var.sec_log_analytics_workspace_id
+  storage_account_id         = var.sec_storage_id
```

Comment:
I don't see the definition of the variable **sec_storage_id**.
It needs to be defined in the variable.tf file of the module.